### PR TITLE
AV-41204: Refactored the code to re-use transport and clients. They a…

### PR DESCRIPTION
**Pull request by grastogi23**
…re co-routine safe

### CODE REVIEW

File: go/session/avisession.go [hunk 1]
```diff
@@ -721,20 +715,14 @@ func (avisess *AviSession) CheckControllerStatus() (bool, error){
...
-
 	//This is an infinite loop. Generating http request for a login URI till controller is in up state.
 	for {
 		check_req, err := http.NewRequest("GET", url, nil)
 		if err != nil {
 			glog.Errorf("CheckControllerStatus Error while generating http request.")
 			return false, err
 		}
-
+		client := &http.Client{Transport: avisess.tr}
```
Comment (ypraveen) [inline go/session/avisession.go]: why create again? why not use avisess.client?

Comment (grastogi23) [reply go/session/avisession.go]: fixed

**Review by grastogi23 — COMMENTED** [review]
fixed the code to remove client re-create

**Review by gitshrikant — APPROVED** [review]

**Comment by grastogi23** [comment]
@ypraveen  ping

**Review by ypraveen — APPROVED** [review]

**Pull request merged**